microsoft-edge/webview2/how-to/add-host-object-to-script.md

```diff
@@ -0,0 +1,295 @@
+---
+title: Use app objects in JavaScript
+description: Use host object in JavsScript using the WebView2 AddHostObjectToScript API.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 2/17/2022
+---
+# Use host objects in JavaScript
+
+WebView2 enables applications to bridge between web and native sides of an application using an object that can be passed between the two sides. Such objects are defined in the native code and often called host objects. They can be projected into JavaScript using a WebView2 API called `AddHostObjectToScript`, as described in this document.
+
+Why use `AddHostObjectToScript`?
+
+  * Developers often encounter a native object whose methods or properties they want to call from JavaScript. They may want these native object methods to be triggered by web code or by user interaction on the web side.
+
+  * In addition, developers may not want to re-implement native object functions, and this API enables re-use of code. For example, there might be a webcam native API, which would take a large amount of code to re-write on the web side. Having the ability to call the native object is quicker and more efficient than re-coding the function. In this case you can pass the object to the web side to reuse the native API function.
```

Comment:
```suggestion
  * In addition, you might not want to re-implement native object functions, and this API enables re-use of code. For example, there might be a webcam native API, which would take a large amount of code to rewrite on the web side. Having the ability to call the native object is quicker and more efficient than re-coding the function. In this case, you can pass the object to the web side, so that the JavaScript code can reuse the native API function.
```